setup.sh

```diff
@@ -42,70 +43,94 @@ fi
+                echo "Finished cleaning $1"
         else
                 echo "$1 .terraform not found"
         fi
 }
 
+import_shared_dev_route53 () {
+# Imports into terraform the shared development route 53 zone that had not been set up
+# using Terraform
+        echo "import shared dev route53"
+        aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH import aws_route53_zone.shared_dev_subdomain $SHARED_DEV_DNS_ZONE
+}
+
+remove_shared_dev_route53 () {
+# Remove the shared development route 53 zone from the state file
+        echo "import shared dev route53"
+        aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH state rm aws_route53_zone.shared_dev_subdomain
+}
+
 init () {
+# Init a terraform project
         echo $1
 
         cd $TERRAFORMPROJ$1
-        if [ "${USE_AWS_VAULT}" = 'true' ] ; then
-                aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH init -backend-config=$TERRAFORMBACKVARS
-        else
-                $TERRAFORMPATH init -backend-config=$TERRAFORMBACKVARS
-        fi
+        aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH init -backend-config=$TERRAFORMBACKVARS
 }
 
 plan () {
+# Plan a terraform project
         echo $1
+
         cd $TERRAFORMPROJ$1
-        if [ "${USE_AWS_VAULT}" = 'true' ] ; then
-                aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH plan --var-file=$TERRAFORMTFVARS
-        else 
-                $TERRAFORMPATH plan --var-file=$TERRAFORMTFVARS
-        fi
+        aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH plan --var-file=$TERRAFORMTFVARS
 }
 
 apply () {
+# Apply a terraform project
         echo $1
 
         cd $TERRAFORMPROJ$1
-        if [ "${USE_AWS_VAULT}" = 'true' ] ; then
-                aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH apply --var-file=$TERRAFORMTFVARS --auto-approve
-        else
-                $TERRAFORMPATH apply --var-file=$TERRAFORMTFVARS --auto-approve
+
+        # For development stacks, for the `infra-networking-route53` project, ensure that
+        # the shared_dev_route53 resource has been imported into terraform before applying
+        if [ $ENV != 'production' -a $ENV != 'staging' -a "$1" = 'infra-networking-route53' ] ; then
+                import_shared_dev_route53
         fi
+
+        aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH apply --var-file=$TERRAFORMTFVARS --auto-approve
 }
 
 destroy () {
+# Destroy a terraform project
         echo $1
 
         cd $TERRAFORMPROJ$1
-        if [ "${USE_AWS_VAULT}" = 'true' ] ; then
-                aws-vault exec ${PROFILE_NAME} -- $TERRAFORMPATH destroy --var-file=$TERRAFORMTFVARS --auto-approve
-        else
-                $TERRAFORMPATH destroy --var-file=$TERRAFORMTFVARS
+
+        # For development stacks, for the `infra-networking-route53` project,
+        # remove the shared_dev_subdomain route53 state file
+        if [ $ENV != 'production' -a $ENV != 'staging' -a "$1" = 'infra-networking-route53' ] ; then
+                remove_shared_dev_route53
```

Comment:
if I run `terraform destroy` manually, am I likely to delete the route 53 zone because I forget to unimport the shared zone?  Is there a way we can avoid this hazard?

Reply:
It doesn't destroy the shared `dev.gds-reliability.engineering` zone as terraform recognises that there are other resources in the zone. 

The un-import should happen automatically for dev stacks, and should allow the resource specific for a user to be removed without affecting other dev stacks.